src/sentry/api/event_search.py

```diff
@@ -405,41 +411,41 @@ def process_list(self, first, remaining):
         ]
 
     def visit_numeric_filter(self, node, children):
-        (search_key, _, (value,)) = children
-        operator = value[0]
+        (search_key, _, operator, search_value) = children
         if isinstance(operator, Node):
-            if isinstance(operator.expr, Optional):
-                operator = "="
-            else:
-                operator = operator.text
+            operator = "=" if isinstance(operator.expr, Optional) else operator.text
         else:
             operator = operator[0]
 
-        if operator == "[":
-            operator = "IN"
-            search_value = self.process_list(value[1], value[2])
+        if self.is_numeric_key(search_key.name):
+            try:
+                search_value = SearchValue(parse_numeric_value(*search_value.match.groups()))
+            except InvalidQuery as exc:
+                raise InvalidSearchQuery(str(exc))
+            return SearchFilter(search_key, operator, search_value)
         else:
-            search_value = value[1]
+            search_value = search_value.text
+            search_value = SearchValue(operator + search_value if operator != "=" else search_value)
+            operator = "=" if operator != "=" else operator
```

Comment:
Seems like `operator` will always be `=` here

Reply:
oh yea lol